### ISSUE TITLE
[tool_crash] ClientException

### ISSUE THREAD

**Issue by RhickMercado** [flutter/flutter#89945]
## Command
```
flutter --no-color run --machine --track-widget-creation --device-id=emulator-5554 --start-paused --dart-define=flutter.inspector.structuredErrors=true --devtools-server-address=http://127.0.0.1:9100 lib/main.dart
```

## Steps to Reproduce
1. ...
2. ...
3. ...

## Logs
ClientException
```
#0      IOClient.send.<anonymous closure> (package:http/src/io_client.dart:49:13)
#1      _invokeErrorHandler (dart:async/async_error.dart:45:24)
#2      _HandleErrorStream._handleError (dart:async/stream_pipe.dart:272:9)
#3      _ForwardingStreamSubscription._handleError (dart:async/stream_pipe.dart:157:13)
#4      _HttpClientResponse.listen.<anonymous closure> (dart:_http/http_impl.dart:712:16)
#5      _rootRunBinary (dart:async/zone.dart:1378:47)
#6      _CustomZone.runBinary (dart:async/zone.dart:1272:19)
#7      _CustomZone.runBinaryGuarded (dart:async/zone.dart:1178:7)
#8      _BufferingStreamSubscription._sendError.sendError (dart:async/stream_impl.dart:360:15)
#9      _BufferingStreamSubscription._sendError (dart:async/stream_impl.dart:378:16)
#10     _BufferingStreamSubscription._addError (dart:async/stream_impl.dart:280:7)
#11     _ForwardingStreamSubscription._addError (dart:async/stream_pipe.dart:128:11)
#12     _addErrorWithReplacement (dart:async/stream_pipe.dart:176:8)
#13     _HandleErrorStream._handleError (dart:async/stream_pipe.dart:277:11)
#14     _ForwardingStreamSubscription._handleError (dart:async/stream_pipe.dart:157:13)
#15     _rootRunBinary (dart:async/zone.dart:1378:47)
#16     _CustomZone.runBinary (dart:async/zone.dart:1272:19)
#17     _CustomZone.runBinaryGuarded (dart:async/zone.dart:1178:7)
#18     _BufferingStreamSubscription._sendError.sendError (dart:async/stream_impl.dart:360:15)
#19     _BufferingStreamSubscription._sendError (dart:async/stream_impl.dart:378:16)
#20     _BufferingStreamSubscription._addError (dart:async/stream_impl.dart:280:7)
#21     _SyncStreamControllerDispatch._sendError (dart:async/stream_controller.dart:737:19)
#22     _StreamController._addError (dart:async/stream_controller.dart:615:7)
#23     _StreamController.addError (dart:async/stream_controller.dart:569:5)
#24     _HttpParser._reportBodyError (dart:_http/http_parser.dart:1183:22)
```
```
[✓] Flutter (Channel stable, 2.2.3, on macOS 11.5.2 20G95 darwin-x64, locale en-PH)
    • Flutter version 2.2.3 at /Users/rhickmercado/Documents/Programming/flutter
    • Framework revision f4abaa0735 (2 months ago), 2021-07-01 12:46:11 -0700
    • Engine revision 241c87ad80
    • Dart version 2.13.4

[✓] Android toolchain - develop for Android devices (Android SDK version 31.0.0)
    • Android SDK at /Users/rhickmercado/Library/Android/sdk
    • Platform android-31, build-tools 31.0.0
    • Java binary at: /Applications/Android Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 11.0.10+0-b96-7281165)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 12.5.1, Build version 12E507
    • CocoaPods version 1.11.0

[✗] Chrome - develop for the web (Cannot find Chrome executable at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome)
    ! Cannot find Chrome. Try setting CHROME_EXECUTABLE to a Chrome executable.

[✓] Android Studio (version 2020.3)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.10+0-b96-7281165)

[✓] IntelliJ IDEA Community Edition (version 2020.3.3)
    • IntelliJ at /Applications/IntelliJ IDEA CE.app
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart

[✓] VS Code (version 1.60.0)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.26.0

[✓] Connected device (2 available)
    • Android SDK built for x86 64 (mobile) • emulator-5554                        • android-x64 • Android 5.1.1 (API 22) (emulator)
    • iPhone 12 Pro Max (mobile)            • EBB9F080-6C57-4CE5-ADA9-6D523BFB5F24 • ios         • com.apple.CoreSimulator.SimRuntime.iOS-14-5 (simulator)

! Doctor found issues in 1 category.

```

## Flutter Application Metadata
**Type**: app
**Version**: 1.0.0+1
**Material**: true
**Android X**: false
**Module**: false
**Plugin**: false
**Android package**: null
**iOS bundle identifier**: null
**Creation channel**: stable
**Creation framework version**: f4abaa0735eba4dfd8f33f73363911d63931fe03
### Plugins
audioplayers-0.19.1
firebase_auth-3.1.1
firebase_auth_web-3.1.0
firebase_core-1.6.0
firebase_core_web-1.1.0
geolocator-7.6.0
geolocator_android-1.0.0
geolocator_apple-1.2.0
geolocator_web-2.0.6
path_provider-2.0.3
path_provider_linux-2.0.2
path_provider_macos-2.0.2
path_provider_windows-2.0.3

**Comment by RhickMercado** [comment]
I Changed my Emulator from Android 4.3 to Android 5.1 and it works. This happened after including Firebase. Did Flutter with Firebase do not have support Android 4.3 any more?

**Comment by darshankawar** [comment]
@RhickMercado 
This issue was fixed and merged into master, #73356
Please switch to latest master and try the same command again to see if it works or issue still persists.

Closing this for now as this is already fixed. If after switching to latest master you see the same issue again, write in comments and I'll reopen it.
